src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -2312,6 +2311,63 @@ public URL createCrumbedUrl(String relativePath) throws IOException {
         }
 
         /**
+         * Add the "Authorization" header with Basic credentials derived from login and password using Base64
+         * @since TODO
+         */
+        public @NonNull WebClient withBasicCredentials(@NonNull String login, @NonNull String passwordOrToken) {
+            String authCode;
+            try {
+                authCode = new String(Base64.encode((login + ":" + passwordOrToken).getBytes("UTF-8")));
+            } catch (UnsupportedEncodingException e) {
+                // impossible
+                throw new Error(e);
+            }
+
+            addRequestHeader(HttpHeaders.AUTHORIZATION, "Basic " + authCode);
+            return this;
+        }
+
+        /**
+         * Use {@code loginAndPassword} as login AND password, especially useful for {@link DummySecurityRealm}
+         * Add the "Authorization" header with Basic credentials derived from login using Base64
+         * @since TODO
+         */
+        public @NonNull WebClient withBasicCredentials(@NonNull String loginAndPassword){
+            return withBasicCredentials(loginAndPassword, loginAndPassword);
+        }
+
+        /**
+         * Retrieve the {@link ApiTokenProperty} from the user, derive credentials from it and place it in Basic authorization header
+         * @see #withBasicCredentials(String, String)
+         * @since TODO
+         */
+        public @NonNull WebClient withBasicApiToken(@NonNull User user){
+            return withBasicCredentials(user.getId(), user.getProperty(ApiTokenProperty.class).getApiToken());
+        }
+
+        /**
+         * Retrieve the {@link ApiTokenProperty} from the associated user, derive credentials from it and place it in Basic authorization header
+         * @see #withBasicApiToken(User)
+         * @since TODO
+         */
+        public @NonNull WebClient withBasicApiToken(@NonNull String userId){
+            User user = User.getById(userId, false);
+            assertNotNull("The userId must correspond to an already created User", user);
+            return withBasicApiToken(user);
+        }
+
+        /**
+         * Remove the "Authorization" header from the client
+         * Meant to be used after having configured the authentication using a token
+         * @see #withBasicCredentials(String, String)
+         * @since TODO
+         */
+        public @NonNull WebClient removeBasicAuthorizationHeader(){
```

Comment:
Probably unnecessary (you can just `createWebClient` again), but OK.

Reply:
You convinced me, I was not satisfied with the name btw.